lib/fastlane/plugin/wpmreleasetoolkit/actions/android/android_generate_po_file_from_metadata.rb

```diff
@@ -0,0 +1,101 @@
+require 'fastlane/action'
+require 'gettext/po'
+
+module Fastlane
+  module Actions
+    class AndroidGeneratePoFileFromMetadataAction < Action
+      REQUIRED_KEYS = %w[full_description title short_description release_notes_short release_notes release_notes_previous].freeze
+      # TODO: release_notes_previous.txt must be generated by the code_freeze lanes
+      SPECIAL_KEYS = %w[release_notes release_notes_previous].freeze
+      KEYS_TO_COMMENT_HASH = {
+        release_notes: 'Release notes for this version to be displayed in the Play Store. Limit to 500 characters including spaces and commas!',
+        full_description: 'Multi-paragraph text used to display in the Play Store. Limit to 4000 characters including spaces and commas!',
+        title: 'Title to be displayed in the Play Store. Limit to 30 characters including spaces and commas!',
+        short_description: 'Short description of the app to be displayed in the Play Store. Limit to 80 characters including spaces and commas!'
+      }.freeze
+
+      def self.run(params)
+        @metadata_directory = params[:metadata_directory]
+        @release_version = params[:release_version]
+
+        prefix = 'play_store'
+        all_keys = Dir[File.join(@metadata_directory, '*.txt')]
+
+        # Remove from all_keys the special keys as they need to be treated specially
+        standard_files = []
+        all_keys.each do |key|
+          standard_files.append(key) unless SPECIAL_KEYS.include? File.basename(key, '.txt')
```

Comment:
@AliSoftware  

here I am assuming that `params[:metadata_directory]` can also contain files other than the required files in `REQUIRED_KEYS`.

If I only need to look for `REQUIRED_KEYS` files in `params[:metadata_directory]` (thus ignoring other files that could be in there) then I could simplify my code quite a bit.

Reply:
It depends on your plan to deal with (and the approach you want to take on how) other sources like `screenshots_*.txt` files.

Those kind of other sources needs to be included in the `.po` if present, and in fact if you remember the your task's P2/request, one of the key issues we have had multiple with the old implementation was a that if new keys/metadata were added (eg new screenshot strings), we often had those missed and not included in the final `.po` because we forgot to sync the list of `.txt` files in the folder with the action's call site listing them explicitly.

Given those cards, and what was discussed in the RfC P2 about the API design ideas for that action, it feel pretty clear to me that it makes more sense to include _every_ `*.txt` file by default here, to solve the very core of our original problem this task was initially started for, namely making sure no txt file failed to be included in the po and failed to be send for translation in the first place.

Reply:
> It depends on your plan to deal with (and the approach you want to take on how) other sources like `screenshots_*.txt` files, so that'd be for you to decide if those extra files needs to be listed explicitly (via some `other_sources` parameter) or would be discovered implicitly. (That was the whole point of the RfC to discuss those kind of points and the API 😉)
> 
> In my opinion and given the latest discussions tho, those kind of other sources should probably be included in the `.po` implicitly if present in the folder. In fact if you remember the your task's P2/request, one of the key issues we have had multiple times with the old implementation, was a that if new metadata files were added (eg new screenshot strings), we often had those missed and not included in the final `.po` because we forgot to sync the list of `.txt` files in the folder with the action's call site listing them explicitly.
> 
> Given that, and what was discussed already around the API design ideas for that action, it makes more sense to me to include _every_ `*.txt` file by default implicitly here, to solve the very core of our original problem this task was initially started for, namely making sure no txt file failed to be included in the po and failed to be send for translation in the first place.

That all makes perfect sense. I would still add an `other_sources` parameter to allow for keeping `screenshots_*.txt` where they already are. Note how `source_metadata_folder` and `custom_metadata_folder` differ.  With this feature there are 2 possibilities:

1. Use `other_sources=[custom_metadata_folder]' in the API to add `screenshots_*.txt`
2. Move all `screenshots_*.txt` into the now unique folder `source_metadata_folder`